wallet/src/service/Offers.ts

```diff
@@ -33,12 +36,40 @@ export type PurseInfo = {
   denom?: string;
 };
 
+// XXX better name?
+type PurseDisplayInfo = {
+  value?: number | string; // Localstorage cannot serialize BigInt.
+  pursePetname?: Petname;
+  amount?: CapData<string>;
+};
 type GiveOrWantEntries = {
-  [keyword: string]: {
-    value?: number | string; // Localstorage cannot serialize BigInt.
-    pursePetname?: Petname;
-    amount?: CapData<string>;
-  };
+  [keyword: string]: PurseDisplayInfo;
+};
+
+const CapDataShape = harden({ body: M.string(), slots: M.array() });
+
+const getBrandPetNameAndDisplayInfoWithoutPurse = async (
+  watcher: ChainStorageWatcher,
+  brand: Brand,
+) => {
+  const brandsP = watcher.queryOnce<[string, unknown][]>([
```

Comment:
This used to be in the top-level of offer service so it's queried one time, but now it's queried for every proposal entry. Consider moving it back so it can be memoized.